tests/release_tests/test_launch/test_kubernetes.py

```diff
@@ -0,0 +1,187 @@
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "aws-secret-access-key",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                    {
+                                        "name": "AWS_SESSION_TOKEN",
+                                        "valueFrom": {
+                                            "secretKeyRef": {
+                                                "name": "aws-session-token",
+                                                "key": "password",
+                                            }
+                                        },
+                                    },
+                                ]
+                            }
+                        ]
+                    }
+                }
+            },
+        }
+    },
+}
+
+
+def test_kubernetes_agent_on_local_process():
+    entity = "launch-release-testing"
+    project = "release-testing"
+    queue = "kubernetes-queue"
+    job_name = "job-source-release-testing-train.py:v1"
+
+    try:
+        # Start launch agent
+        agent_process = run_cmd_async(f"wandb launch-agent -q {queue} -e {entity}")
+
+        # Start run
+        queued_run = launch_add(
+            job=f"{entity}/{project}/{job_name}", queue_name=queue, entity=entity
+        )
+
+        # Assert successful
+        run = queued_run.wait_until_finished()
+        assert run.state == "finished"
+    finally:
+        agent_process.kill()
+        run_cmd("rm -r artifacts")
+
+
+@pytest.mark.timeout(180)
+def test_kubernetes_agent_in_cluster(api_key, base_url):
+    entity = "launch-release-testing"
+    project = "release-testing"
+    queue = "kubernetes-queue"
+    job_name = (
+        "job-305054156030.dkr.ecr.us-east-2.amazonaws.com_release-testing_latest:v0"
+    )
+
+    # get user's wandb API key and AWS creds
+    if not api_key:
+        api_key = get_wandb_api_key(base_url)
+    aws_id = os.getenv("AWS_ACCESS_KEY_ID")
+    aws_secret = os.getenv("AWS_SECRET_ACCESS_KEY")
+    aws_token = os.getenv("AWS_SESSION_TOKEN")
+    _create_wandb_and_aws_secrets(api_key, aws_id, aws_secret, aws_token)
+
+    run_cmd("kubectl apply -f tests/release_tests/test_launch/launch-config.yml")
+    run_cmd("kubectl apply -f tests/release_tests/test_launch/launch-agent.yml")
+
+    # Capture sigint so cleanup occurs even on ctrl-C
+    sigint = signal.getsignal(signal.SIGINT)
+
+    def cleanup(signum, frame):
+        signal.signal(signal.SIGINT, sigint)
+        _cleanup_deployment()
+        sys.exit(1)
+
+    signal.signal(signal.SIGINT, cleanup)
+
+    pod_name = None
+    try:
+        # Start run
+        launch_add(
```

Comment:
launch_add returns a queued_run object, which we can assert on right? maybe we add just one or two lines after completion making sure the run actually finished and it was healthy. Not sure what this job does, but it would be pretty easy to have it log some trivial stuff and then confirm with the public API the subsequent run contains those logs in the summary.